core/src/main/java/dk/alexandra/fresco/framework/util/HmacDeterministicRandomBitGeneratorImpl.java

```diff
@@ -0,0 +1,129 @@
+package dk.alexandra.fresco.framework.util;
+
+import dk.alexandra.fresco.framework.MPCException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.function.Supplier;
+import javax.crypto.Mac;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * This class implements the HMac-DRBG (HMac-Deterministic Random Bit Generator) as specified in
+ * https://csrc.nist.gov/publications/detail/sp/800-90a/rev-1/final
+ * 
+ * <p>
+ * Note that this class is not threadsafe.
+ * </p>
+ * <p>
+ * Note also that reseeding (calling {@link #setSeed(byte[])}) should occur at least every 2^42
+ * calls to {@link #nextBytes(byte[])} to be secure. No signal will be send about this.
+ * </p>
+ */
+public class HmacDeterministicRandomBitGeneratorImpl implements DeterministicRandomBitGenerator {
```

Comment:
Please, let's call this HmacDrbg (certainly drop the Impl suffix when there is not an overlap with the interface name).